utility/detailshelper.php

```diff
@@ -36,7 +36,8 @@ public function getDetails($fileId, $userFolder) {
 		$fileNodes = $userFolder->getById($fileId);
 		if (\count($fileNodes) > 0) {
 			$data = $this->extractor->extract($fileNodes[0]);
-
+			array_walk($data['audio'], [$this, 'sanitize_string']);
+			array_walk($data['comments'], [$this, 'sanitize_string']);
```

Comment:
This call probably does nothing because $data['comments'] does not contain any strings, it contains arrays. That's why there's the self::flattenComments call below.

Reply:
I have updated the order of the operations to solve this issue.

Reply:
Thanks. I have two more minor stylistic requests.
1. Instead of `array_walk($something, [$this, 'sanitize_string'])`, you could achieve the same with `array_walk($something, 'self::sanitize_string')`. I feel this would be slightly cleaner.
2. All the PHP codes of the Music app use camelCase instead of snake_case. So the function name should actually be `sanitizeString`.

Reply:
You're right. Updated for consistent style.

Reply:
The inspection doesn't understand that the method is called with this code:
`array_walk($something, 'self::sanitize_string')`
I would suggest to revert to the original one then? `array_walk($something, [$this, 'sanitize_string'])`

Reply:
I'll let you decide, I'm content either way. On one hand, it's good to avoid creating Scrutinizer warnings. On the other hand, I dislike a bit the idea that we let false positives from an analysis tool to control our coding.

Reply:
scrutinizer doesn't really detect it correctly, even with the full class name. so I suggest to keep it as it is now.